tests/integration/DistributedCacheInvalidationTest.php

```diff
@@ -0,0 +1,280 @@
+
+        // Wait for throttle interval to pass
+        sleep(6);
+
+        // Second request - middleware should detect the change
+        $response2 = $this->send($this->request('GET', '/'));
+        $this->assertEquals(200, $response2->getStatusCode());
+
+        // Verify version was updated in Redis
+        $currentVersion = $redis->connection('fof.cache')->get('flarum:cache:version');
+        $this->assertEquals($newVersion, (int) $currentVersion);
+    }
+
+    /**
+     * @test
+     */
+    public function local_file_caches_are_invalidated_when_middleware_detects_version_change()
+    {
+        $this->requiresRedis();
+
+        $paths = $this->app()->getContainer()->make(Paths::class);
+        $redis = $this->app()->getContainer()->make(Factory::class);
+        $middleware = $this->app()->getContainer()->make(DistributedCacheInvalidation::class);
+
+        // Create test cache files
+        $formatterCache = $paths->storage.'/formatter';
+        $localeCache = $paths->storage.'/locale';
+        $viewsCache = $paths->storage.'/views';
+
+        // Ensure directories exist
+        @mkdir($formatterCache, 0755, true);
+        @mkdir($localeCache, 0755, true);
+        @mkdir($viewsCache, 0755, true);
+
+        // Create test files
+        file_put_contents($formatterCache.'/test.php', '<?php // test');
+        file_put_contents($localeCache.'/en.php', '<?php return [];');
+        file_put_contents($viewsCache.'/test.blade.php', '<html></html>');
+
+        // Verify files exist
+        $this->assertFileExists($formatterCache.'/test.php');
+        $this->assertFileExists($localeCache.'/en.php');
+        $this->assertFileExists($viewsCache.'/test.blade.php');
+
+        // Set a cache version in Redis
+        $redis->connection('fof.cache')->set('flarum:cache:version', time());
+
+        // Directly invoke the middleware logic via reflection to simulate version detection
+        // Note: In real PHP-FPM workers, the middleware detects version changes across requests
+        // In tests with process isolation, we test the invalidation logic directly
+        $reflectionClass = new \ReflectionClass($middleware);
+
+        $shouldInvalidateMethod = $reflectionClass->getMethod('shouldInvalidateLocal');
+        $shouldInvalidateMethod->setAccessible(true);
+
+        $invalidateMethod = $reflectionClass->getMethod('invalidateLocalCaches');
+        $invalidateMethod->setAccessible(true);
+
+        // Call invalidation directly
+        $invalidateMethod->invoke($middleware);
+
+        // Files should be cleared
+        $this->assertFileDoesNotExist($formatterCache.'/test.php', 'test.php should be deleted');
+        $this->assertFileDoesNotExist($localeCache.'/en.php', 'en.php should be deleted');
+        $this->assertFileDoesNotExist($viewsCache.'/test.blade.php', 'test.blade.php should be deleted');
+    }
+
+    /**
+     * @test
+     */
+    public function middleware_throttles_redis_checks()
```

Comment:
Unsure about this test. I would test this by setting the `distributed_invalidation_interval` to a high number and running an adjusted version of `local_file_caches_are_invalidated_when_middleware_detects_version_change` where the test files _should_ still exist after a delay. This would prove that
- throttling works
- overwriting the `DEFAULT_CHECK_INTERVAL` is possible